### DISCUSSION TITLE
Safe voltage on analog input

### DISCUSSION THREAD

**Discussion by Pandagineer** [bitcraze/discussions#323] [Q&A] [answered]
Hello,

I want to measure analog voltage. Reading the Breakout Deck spec sheet (https://www.bitcraze.io/documentation/hardware/breakout_deck/breakout_deck-datasheet.pdf), I find analog input channels A0 thru A4 (pink color). 

1. Where should my ground go? There is the GND pin, but this is colored orange (power) in the spec sheet, so I want to make it is also ground for the analog inputs.
2. What is a safe voltage limit for the analog inputs?

Thanks!

**Comment by tobbeanton** [comment, marked as answer]
1. Yes use the pin marked GND
2. 3.0V is the upper limit for the analog inputs (with some margin).

**Reply by Pandagineer** [reply]
Thank you. Just so I'm clear on margin: I should stay away from 3.0 V (e.g. stay below 2.7V), or I can go a little over 3.0 V (e.g. stay below 3.2 V)?

**Reply by tobbeanton** [reply]
Depending on how you configure the ADC you can get different effects but most common way to do it would be to use Vref as max input. Thus you don't want to go above 3.0V as this would just read out as max (e.g. 4096 if 12bits are used). I recommend you to read up on e.g. a "STM32 ADC tutorial). Same thing if you would use the analogReadVoltage() function, saturate at 3.0V, which will do the conversion to voltage for you.

**Reply by Pandagineer** [reply]
Roger. Thanks
